docs/testplan/DHCP-Server-test-plan.md

```diff
@@ -0,0 +1,83 @@
+# DHCP Server Test Plan
+
+- [Overview](#overview)
+  - [Scope](#scope)
+  - [Testbed](#testbed)
+- [Setup configuration](#setup-configuration)
+- [Test Cases](#test-cases)
+  - [Server/Relay test case](#serverrelay-test-case)
+  - [Default test case](#default-test-case)
+  - [Emulate port change](#emulate-port-change)
+  - [Multi-Vlan](#multi-vlan)
+
+## Overview
+
+The purpose is to test the functionality of dnsmasq DHCP Server on SONiC switch DUT.
+
+## Scope
+
+The test is targeting a running SONiC system with dhcp server configured.
+Purpose of the test is to verify a DHCP Server on SONiC successfully distributed expected IP addresses to clients
+
+## Testbed
+
+Supported topologies: t0, t0-52
+
+## Setup configuration
+
+    1. Configure DHCP server option in config_db. DHCP Relay will be disabled if DHCP server is enabled.
+	2. Start dnsmasq using dnsmasq -p 0 --conf-dir=/etc/dnsmasq.d/ -z. 
+	3. Update desired subnet ip in updateHostAddresses.py. Start the python script using updateHostAddresses.py.
+
+## Test cases
+
+## Server/Relay test case
+
+If dhcp_server flag is enabled:
+- dnsmasq is enabled
+- dhcrelay is disabled
```

Comment:
dhc**p**relay?

Reply:
dhcrelay is the process name for isc-dhcpv4. Shall I change to DHCPv4 Relay for easier understanding?